scvi/model/base/_vaemixin.py

```diff
@@ -210,3 +213,45 @@ def get_latent_representation(
             if return_dist
             else torch.cat(latent).numpy()
         )
+
+    @torch.inference_mode()
+    def get_batch_representation(
+        self,
+        adata: AnnData | None = None,
+        batch_keys: list[str] | None = None,
+    ) -> np.ndarray:
+        """Return batch representations.
+
+        Parameters
+        ----------
+        adata
+            AnnData object with equivalent structure to initial AnnData. If ``None``, defaults to
+            the AnnData object used to initialize the model.
+        batch_keys
+            Keys for batches to return representations for. If ``None``, all batches are used.
+
+        Returns
+        -------
+        Array of batch representations of shape ``(n_batches, batch_embedding_dim)``.
+        """
+        self._check_if_trained(warn=False)
+
+        if not hasattr(self.module, "batch_embedding"):
+            raise NotImplementedError("Model does not support batch embeddings.")
+        if self.module.batch_embedding is None:
+            raise ValueError("Model was not trained with batch embeddings.")
+
+        adata = self._validate_anndata(adata)
```

Comment:
Add a warning if batch_key has similar name but actually describes different batches the result will be non-sense.

Reply:
Thanks for taking a look! I'm not following, could you clarify what you mean?

Reply:
If a user provides a new adata object with different batches, in which batches are e.g. just numbered, it will output batch embeddings for those that have no meaning. I think it's important that we allow using a new adata object but we could issue a warning.

Reply:
I guess the same thing would happen without batch_embeddings as well. Marking it as resolved.

Reply:
Ah I see what you mean now. I have to think more about this.

Reply:
I feel like the simplest solution to this would be removing the `adata` argument in `get_batch_representation`.